### ISSUE TITLE
[Bug] Clip overlay is doubled when rotating the phone back and forth

### ISSUE THREAD

**Issue by radu-andrei-oprea** [futo-org/grayjay-android#582]
Version 221 - when rotating the phone, the clip overlay might get doubled, from the progress bar to the top buttons on the clip window. Everything except for the play/pause button and the title. The new buttons/bar appear slightly top right of their original positions. The new buttons directly overlap the original ones, but function as intended. The new bar functions as intended as well, and is in sync with the old one. Both can be used to move through the video.

This will happen either when flipped vertically or horizontally. It will fluctuate between normal and doubled, when rotating one or more times.

Behaviour is inconsistent. A way to recreate it every time has not been yet pinned down. Suspected cause might be watching a youtube video with a slightly longer title. However, after closing and opening the app again, the bug still persists.

Device: Mi 10T Pro 5G
MiUI Global 13.0.12
Android 12 skq1.211006.001

**Comment by IUEngineering** [comment]
Have the same issue with a OnePlus Nord 2T 5G

**Comment by xephos1one** [comment]
Here are some screenshots in landscape and portrait mode:

![Screenshot_20231231-112541](https://github.com/futo-org/grayjay-android/assets/39647898/20f9310b-6cae-46c0-89e8-5a75ce1127f3)
![Screenshot_20240101-131252](https://github.com/futo-org/grayjay-android/assets/39647898/3a431751-6a06-4bc7-82e3-f4d9c92899a3)

**Comment by koen-futo** [comment]
Possible fix implement which will be deployed in the next update.

**Comment by radu-andrei-oprea** [comment]
> Possible fix implement which will be deployed in the next update. I have seen it before but not been steadily able to reproduce it. Therefore I was only able to make a theoretical fix.

One week later, I haven't encountered the issue anymore. If it's ok with you, I can close the issue.

**Comment by xephos1one** [comment]
Still happening to me in version 227

**Comment by koen-futo** [comment]
@xephos1one are you able to give any more details?

**Comment by xephos1one** [comment]
@koen-futo It took some time to find a possible way to replicate it a few times:
- Have a video (I only tested withYoutube) paused in landscape mode
- Lock the screen while in landscape mode
- Unlock the screen in portrait mode
- Turn the phone back to landscape

Sadly it only works sometimes.

**Comment by radu-andrei-oprea** [comment]
@koen-futo okay, it came back. @xephos1one might have a point, it definitely has something to do with locking the screen in landscape mode.

I also found a way to trigger this (although it's inconsistent):

1. Lock the phone in landscape mode while watching a video (Youtube, although it's likely irrelevant)
2. Unlock the phone horizontally
3. Turn it vertically
4. Press on the screen to show the overlay
5. Flip the phone while the overlay is showing
6. Once in landscape mode, the bug might occur

**Comment by xephos1one** [comment]
Yesterday it also happened directly after opening the app, clicking on a video in portrait mode and turning the phone to landscape mode - without locking the screen.
The timing might be important between opening the video in portrait mode and turning to landscape 🤔

**Comment by Matesaktesak** [comment]
I can't pinpoint what's triggering this issue, but it's happening daily. Running v233 release

**Comment by Dragonjinx** [comment]
Somewhat consistent way of triggering this for me:

- Watch a downloaded video
- auto switch to landscape mode
- manually switch back to portrait mode

**Comment by F1nnM** [comment]
Another data point maybe:
For me it happens without locking the screen or anything.

When rotating the phone, 90% of the time it transitions smoothly into the fullscreen video, and the controls don't duplicate. 10% of the time it briefly flashes the home screen (or whichever one is "under" the video), and when that happens the duplicates appear.

**Comment by donatorsky** [comment]
Hi everyone, wanted to add few cents to the discussion.

It happens to me always while playing any YouTube video when I rotate the phone very quickly and the app switches from/to landscape to/from portrait mode. No additional steps required, no controls visuals must be visible before rotation, etc. I'm attaching (a bit chaotic, but I believe it all gets clear when viewed slowed down) screen recording. Running v236 release on a Samsung Galaxy S24 Ultra with Android 14.

https://github.com/futo-org/grayjay-android/assets/5360041/f42fef77-4e7b-46a8-a95f-4a7f3d75030b

**Comment by Matesaktesak** [comment]
I believe the problem is, that the app is displaying landscape-mode controls even in portrait. I'd guess that some boolean toggle function is sometimes not triggering. I'm so sorry, but I don't have the time to dig through the source to find it...

**Comment by Matesaktesak** [comment]
BUMP: This is still very much a problem, it happens to me every day (mostly). It's not making the app not function, but its annoying.

**Comment by support-grayjay** [comment]
Successfully reproduces on the following devices:

Device 1: Samsung Galaxy Note 20 Ultra (Android 12)
Device 2: Xiaomi Redmi Note 8 Pro (Android 11)
Device 3: Xiaomi 11T Pro (Android 14)

**Device settings:**

Lock orientation off (auto-rotate on)
Dark mode

**Grayjay settings:** (these things might affect the bug):

Auto-Rotate - Same as system
Allow reverse portrait - on
Rotation zone - 30
Stability threshold time - 100

**Reproduction rate:** 2/10

It's really hard to determine what's causing this, but I can confirm that it happens whether you're playing the video from the Home page or if you have the video downloaded and play it from your Downloads. I attached a video where I managed to capture it once — I once also caught it after I locked the screen in landscape mode and unlocked it in portrait mode, but my recording stopped when I locked the phone, so I don't have that video. I think it's clearly having trouble with orientation changes, especially if you spam it. It's possible it has something to do with the exact moment you flip the phone (like right when the controls fade out). It took me a while to recreate it a few times and it's impossible to pinpoint the exact cause because it's really inconsistent.

**Video example:**

https://github.com/user-attachments/assets/7c2e6bc6-6e68-435c-acd8-d541a34f0601

**Comment by Zvonimir-FUTO** [comment]
This should be fixed with the latest rotation changes.

**Comment by Zvonimir-FUTO** [comment]
Any idea how to reproduce it?

**Comment by TheNewsYorkTime** [comment]
I have experienced the original issue with auto-rotate off. It doesn't happen often enough for me to have figured out what the procedure is for me.

**Comment by theFramebuffer** [comment]
I figured out the trigger: turning off and on the phone while in fullscreen.
I'm on a Samsung Galaxy A35 running One UI 8.5

**Comment by theFramebuffer** [comment]
https://github.com/user-attachments/assets/ec47de8a-9392-4219-85d8-1b795b314cb4

CORRECTION: this only happens when i don't turn on the screen then unlock with my fingerprint, but rather ONLY when i turn off my screen then unlock it directly with my fingerprint without turning it back on.

**Comment by Zvonimir-FUTO** [comment]
@erCoso Thank you VERY much for figuring this out. We finally managed to reproduce it and fix it. The fix will be included in next release.

**Comment by theFramebuffer** [comment]
Happy to help! It's about damn time, this bug has been haunting Grayjay for almost 3 years...

**Comment by Zvonimir-FUTO** [comment]
Yeah I hate those bugs so much... I see bunch of people having issues but I can't reproduce it on any of my 5 phones and emulators either...

**Comment by defiling9046** [comment]
@Zvonimir-FUTO should this be marked as completed?

**Comment by Zvonimir-FUTO** [comment]
> [@Zvonimir-FUTO](https://github.com/Zvonimir-FUTO) should this be marked as completed?

Issue will be closed once the fix has been deployed so users don't open new issues thinking we failed to fix the bug and it's "still present" in latest version.
